system/jlib/jhash.hpp

```diff
@@ -49,27 +49,30 @@ interface jlib_decl IAtom : extends IMapping
 {
  public:
     virtual const char * getNamePtr() const = 0;
+    const char *         getAtomNamePtr() const { return IAtom::getAtomNamePtr(this);} // ok if this=NULL
```

Comment:
I would be very happy to remove the duplicate functions, probably sticking with str() for consistency with other classes.

Richard you are correct, it is still technically invalid and a future iteration of the compiler is likely to catch this and report the problem again.

I think the correct solution is to add a new global function e.g.,

inline const char * str(const IAtom * atom) { return atom ? atom->getAtomName() : NULL; }

And change everything to call that.  Unfortunately I can't think of any change that doesn't require large numbers of changes throughout the source code.

If this change is made to master then we should also add the same global function to 5.4, (or even 5.2.6) so that any new code that is written in those branches can call that new function, and not cause problems when it is up-merged.

I'm not sure about the cast operator.  I think it would be fine to keep it, but it should only be used if the IAtom * is not NULL.  That would probably require removing it, checking all places that call it and modifying any where it could be NULL, and then adding it back again.  It may be much simpler, and safer, to remove it completely.